tests/conftest.py

```diff
@@ -2719,6 +2720,57 @@ def user_factory_session(request, htpasswd_identity_provider, htpasswd_path):
     return users.user_factory(request, htpasswd_path)
 
 
+@pytest.fixture(autouse=True)
+def log_alerts(request):
+    """
+    Log alerts at the beginning and end of each test case. At the end of test
+    case print a difference: what new alerts are in place after the test is
+    complete.
+
+    """
+    alerts_before = []
+    prometheus = None
+
+    try:
+        prometheus = PrometheusAPI()
+    except Exception as e:
+        log.warning(f"There was a problem with connecting to Promeheus: {e.message}")
+
+    def _collect_alerts():
+        try:
+            alerts_response = prometheus.get(
```

Comment:
we have seen in the past that in rare occasions, `prometheus.get()` can fail with one of `IndexError` of `ScannerError`.
See https://github.com/red-hat-storage/ocs-ci/blob/c950799f545b3b84d3539302a4b3abd475a7fcce/ocs_ci/ocs/cluster_load.py#L339

Since we this fixture will run for each and every test case, will it make sense to add the same retry decorator here as well?

Reply:
The query slightly differs and I haven't seen this error for this query from monitoring tests. I would rather not put it there because there are times that this should fail (when monitoring is not enabled - although it would probably not fail on index error) and add 15*5 wait for failed query could slow down some test runs too much.

Reply:
Since this is going to be executed for every test to make additional evidence available, I would rather tried once and just report an error in case or problems, instead of trying multiple times and possible extending runtime of the whole suite. Also unless monitoring is disabled, broken or a whole cluster under high load, it should work fine.

Reply:
ack